Jenkinsfile

```diff
@@ -29,6 +30,77 @@ pipeline {
                 sh 'build/run make -j\$(nproc) test-integration'
             }
         }
+
+        stage('Integration Tests SetUp') {
+            steps {
+                sh 'tests/scripts/saveImages.sh > BUILD_NUMBER'
+                script{
+                    BUILD_NUM = readFile('BUILD_NUMBER').trim()
+                }
+                echo "Saved images from build ${BUILD_NUM}"
+                stash name: 'repo',excludes: '_output/,vendor/,.cache/,.work/'
+            }
+        }
+
+        stage('Integration Tests - parallel runs') {
+            environment {
+                KUBE_VERSION = "v1.6"
+            }
+            steps {
+                parallel(
+                    "GCE_K8s_v1.7" : {
+                        node("gce_vm"){
+                            script {
+                                try{
+                                    withEnv(["KUBE_VERSION=v1.7"]){
+                                        deleteDir()
+                                        unstash 'repo'
+                                        sh "docker load -i rookamd64.tar"
+                                        sh "docker load -i toolboxamd64.tar"
+                                        sh 'build/run make -j\$(nproc) build.common'
+                                        sh "tests/scripts/kubeadm-dind.sh up ${BUILD_NUM}"
+                                        sh 'build/run make -j\$(nproc) test-integration'
+                                    }
+                                }
+                                finally{
+                                    archive '_output/tests/*'
```

Comment:
given that our tests make changes at the kernel level (rbd|cephfs) I don't think we can rely on the host machine not getting tainted after a test (especially a bad test or code). Can these agents that run integrations tests be restarted at the VM level between test runs?